### PR TITLE
Move everything to one repository

### PR DESCRIPTION
**Pull request by PerilousApricot**
We currently have many independent repositories we cobble together to make one seamless `lstore` release. Flip it around using [git subtree](https://raw.githubusercontent.com/git/git/master/contrib/subtree/git-subtree.txt) -- this makes one seamless `lstore` repository and lets us continue to split it out into independent repositories For example, there will still be a toolbox repo, but instead of branching and PR-ing that repo, the main repo takes all changes, who can then be pushed back to the independent repos once its stable.

Basically, simplify the fast path (people developing lstore) instead of the slow path (people who want only a checkout of the ibp client library).

### CODE REVIEW

**Comment by PerilousApricot** [comment]
NOTE: If you want to take a look around, use a new clone -- git barfed when I tried to switch from this branch to master and back. YMMV.

**Comment by PerilousApricot** [comment]
Also, ignore the barf of a commit log -- I didn't squash the history of each individual repository, so the amended history for this branch is the sum of all four repositories.

**Comment by PerilousApricot** [comment]
Nearly have all of this ready to commit:
- I had to rewrite history to pretend everything was always in one repo. Jenkins will test to make sure you don't try to re-push the old history back.
- I did a one-off test that the lio repository can be re-split to its own repository. This test resulted in the same hash as the current external repo, so the round trip works right (see [this](https://lostechies.com/johnteague/2014/04/04/using-git-subtrees-to-split-a-repository/) for more info on `git subtree`)
- Since gridftp is all kinda of weird, it's temporarily disabled
